src/mcp.jl

```diff
@@ -84,22 +84,33 @@ function sortperm_MCP_data(obj::Array{ComplementarityType,1})
 end
 
 # Using PATHSolver
-function _solve_path(m::JuMP.Model; linear=false)
+function _solve_path(m::JuMP.Model; linear=false, kwargs...)
```

Comment:
We should lift the `NLPEvaluator` out of `myfunc` and `myjac`. 

```Julia
function _solve_path(m::JuMP.Model; linear = false, kwargs...)
    d = JuMP.NLPEvaluator(m)
    MOI.initialize(d, [:Jac])  # Shouldn't this be Jac not :Grad?
    J_struct = MOI.jacobian_structure(d)
    nnz = length(J_struct)
    I = first.(J_struct)
    J = last.(J_struct)
```

Reply:
Can `MOI.initialize(d, [:Jac])` be also lift out?